src/Configuration.php

```diff
@@ -102,12 +363,87 @@ public function getAppData()
     }
 
     /**
-     * Get the device information.
+     * Set the hostname.
+     *
+     * @param string|null $hostname the hostname
+     *
+     * @return $this
+     */
+    public function setHostname($hostname)
+    {
+        $this->deviceData['hostname'] = $hostname;
+
+        return $this;
+    }
+
+    /**
+     * Get the device data.
      *
      * @return array
      */
     public function getDeviceData()
     {
         return array_merge(['hostname' => php_uname('n')], array_filter($this->deviceData));
     }
+
+    /**
+     * Set custom metadata to send to Bugsnag.
+     *
+     * You can use this to add custom tabs of data to each error on your
+     * Bugsnag dashboard.
+     *
+     * @param array[] $metaData an array of arrays of custom data
+     * @param bool    $merge    should we merge the meta data
+     *
+     * @return $this
+     */
+    public function setMetaData(array $metaData, $merge = true)
```

Comment:
Note the BC break here. Setting merging to be turned on by default to be consistent with the behaviour on the error class.